### DISCUSSION TITLE
clippy::forget-non-drop on #[derive(Bundle)]

### DISCUSSION THREAD

**Discussion by Indy2222** [bevyengine/bevy#5166] [Q&A] [answered]
Since https://github.com/rust-lang/rust-clippy/pull/8630 got into upstream Clippy, it started to complain in my project with:

```
error: call to `std::mem::forget` with a value that does not implement `Drop`. Forgetting such a type is the same as dropping it
  --> crates/spawner/src/draft.rs:31:10
   |
31 | #[derive(Bundle)]
   |          ^^^^^^
   |
   = note: `-D clippy::forget-non-drop` implied by `-D warnings`
note: argument has type `draft::Draft`
  --> crates/spawner/src/draft.rs:31:10
   |
31 | #[derive(Bundle)]
   |          ^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#forget_non_drop
   = note: this error originates in the derive macro `Bundle` (in Nightly builds, run with -Z macro-backtrace for more info)
```

I would be nice if this is somehow addressed so people do not have to turn off this warning themselves.

I did not want to open an issue since I do not know what is the desired state.

**Comment by TimJentzsch** [comment, marked as answer]
This is a known issue, you can find a workaround in <https://github.com/bevyengine/bevy/issues/4601#issuecomment-1114332185> :)

**Reply by Indy2222** [reply]
Thank you for pointing out to the issue! I solved it in the suggested way, i.e. by turning off the lint (in the affected modules). However, I would be more happy if it wasn't necessary.

**Reply by DJMcNab** [reply]
Isn't it already fixed on `main` since #3001?

@alice-i-cecile can you confirm (or anyone else) and close #4601. (I unfortunately cannot check at the moment)

**Reply by alice-i-cecile** [reply]
I'll verify this later today; I have a project with that lint explicitly ignored that I can migrate to main.
